### PR TITLE
docs(button): enhance button docs and examples

### PR DESCRIPTION
**Pull request by EladBezalel**
changed the overview example to
image

and change examples section to 
image

showing use cases

### CODE REVIEW

File: src/material-examples/button-toolbar/button-toolbar-example.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.spacer {
+  flex: 1;
+}
```
Comment (amcdnl) [inline src/material-examples/button-toolbar/button-toolbar-example.css]: Should end w/ new line

File: src/material-examples/button-toolbar/button-toolbar-example.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<mat-toolbar color="primary">
+  <button mat-icon-button>
+    <mat-icon aria-label="Example icon-button with a menu icon">menu</mat-icon>
+  </button>
+  <span>Toolbar with icon buttons</span>
+  <span class="spacer"></span>
+  <button mat-icon-button>
+    <mat-icon aria-label="Example icon-button with a more vertical icon">more_vert</mat-icon>
+  </button>
+</mat-toolbar>
```
Comment (amcdnl) [inline src/material-examples/button-toolbar/button-toolbar-example.html]: Should end w/ new line

File: src/material-examples/button-toolbar/button-toolbar-example.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * @title Icon button usage in toolbar
+ */
+@Component({
+  selector: 'button-toolbar-example',
+  templateUrl: 'button-toolbar-example.html',
+  styleUrls: ['button-toolbar-example.css'],
+})
+export class ButtonToolbarExample {}
```
Comment (amcdnl) [inline src/material-examples/button-toolbar/button-toolbar-example.ts]: Should end w/ new line

**Comment by EladBezalel** [comment]
fixed

File: src/material-examples/button-card/button-card-example.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<mat-card>
+  <mat-card-header>
+    <mat-card-title>Card with action buttons</mat-card-title>
+  </mat-card-header>
+  <mat-card-actions align="end">
+    <button mat-button>CANCEL</button>
```
Comment (jelbourn) [inline src/material-examples/button-card/button-card-example.html]: Use normal text case for buttons, the docs site capitalizes them via CSS. If screen-readers encounter ALL CAPS, it reads it as "A. L. L. C. ..."

(here and elsewhere)

File: src/material-examples/button-overview/button-overview-example.css [hunk 5]
```diff
@@ -1 +1,16 @@
-/** No CSS for this example */
+.button-row {
```
Comment (jelbourn) [inline src/material-examples/button-overview/button-overview-example.css]: Prefix all example styles with `.example-` (here and elsewhere)

This makes it super obvious which styles are from the example vs. the docs site (`docs-`) vs. material

File: src/material-examples/button-card/button-card-example.html [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    <button mat-button color="primary">ACCEPT</button>
```
Comment (jelbourn) [inline src/material-examples/button-card/button-card-example.html]: These should have an `aria-label` along the lines of "Example button, no action"

(here and elsewhere, where the label can be expanded with what the button is an example of)

File: src/material-examples/button-overview/button-overview-example.html [hunk 7]
```diff
@@ -1 +1,76 @@
...
+  <button mat-raised-button color="accent">Accent</button>
+  <button mat-raised-button color="warn">Warn</button>
+  <button mat-raised-button disabled>Disabled</button>
+  <a mat-raised-button routerLink=".">Link</a>
+</div>
+
+<h3>Icon Buttons</h3>
+<div class="button-row">
+  <button mat-icon-button>
+    <mat-icon aria-label="Example icon-button with a heart icon">favorite</mat-icon>
```
Comment (jelbourn) [inline src/material-examples/button-overview/button-overview-example.html]: https://material.angular.io/components/icon/overview#accessibility

In cases like this the `aria-label` should go on the interactive element (the button or anchor)

File: src/material-examples/example-module.ts [hunk 8]
```diff
@@ -17,10 +17,11 @@ import {AutocompleteDisplayExample} from './autocomplete-display/autocomplete-di
 import {AutocompleteFilterExample} from './autocomplete-filter/autocomplete-filter-example';
 import {AutocompleteOverviewExample} from './autocomplete-overview/autocomplete-overview-example';
 import {AutocompleteSimpleExample} from './autocomplete-simple/autocomplete-simple-example';
+import {ButtonCardExample} from './button-card/button-card-example';
```
Comment (jelbourn) [inline src/material-examples/example-module.ts]: Not sure if you knew this but this file is auto-generated

(I'm not sure why it's in source control... @amcdnl?)

**Comment by EladBezalel** [comment]
https://github.com/angular/material.angular.io/pull/330 here's the pr for the [material.angular.io](https://material.angular.io) website

**Comment by EladBezalel** [comment]
@jelbourn it seems like once i removed `example-module.ts` changes it's breaking..

**Comment by jelbourn** [comment]
The changes to example-module are necessary, but you shouldn't need to make them by hand. The file also probably shouldn't need to be in source control (a matter for another PR)

**Comment by jelbourn** [comment]
@EladBezalel could you rebase this PR?

**Comment by josephperrott** [comment]
@EladBezalel could you rebase this PR (again)?

**Comment by josephperrott** [comment]
Closing as we are going on ~6 months of inactvity on this PR.  Feel free to open up a new PR with these changes if you would like to get them into the library.

**Pull request closed without merging**